pkg/api-server/resource_endpoints.go

```diff
@@ -149,6 +157,10 @@ func (r *resourceEndpoints) createOrUpdateResource(request *restful.Request, res
 	}
 }
 
+func (r *resourceEndpoints) validateResourceSchema(resourceRest rest.Resource) error {
+	return plugins.ValidateResourceSchema(resourceRest.GetSpec(), string(r.descriptor.Name))
```

Comment:
can this be called in `Resource#Validate()`?
I really liked that we had this static `Validate()` method that checks the validity of the resource. Now, this method won't check some things because some of the validations is in `Validate()` and some is here.

Reply:
yes, but correct me if I'm wrong - this will cause the schema validation to be triggered on k8s and universal and some people are not a fan of double validation. If we're fine with that then 👍

Reply:
Yes. But assuming that builtin K8S schema validation will fail, we won't even get here.
If we for some reason have a problem that is not caught by K8S, but is caught by this validator - even better for us.

Reply:
done